File: go/problems/day23_test.go
package problems

import "testing"

var day23 Solvable[int] = Day23{}

func TestDay23_TestSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day23_1")
	result := day23.Solve(input)

	var expected int = 94
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

func TestDay23_FullSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day23_2")
	result := day23.Solve(input)

	var expected int = 2362
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

File: ts/problems/day23.test.ts
import test, { describe } from "node:test";
import assert from "node:assert";

import solve from "./day23.js";
import { readProblemInput } from "../inputReader.js";

describe("Day 23", () => {
  test("test set should yield expected result", () => {
    const input = readProblemInput("day23_1");
    const result = solve(input);
    assert.equal(result, 94);
  });

  test("full set should yield expected result", () => {
    const input = readProblemInput("day23_2");
    const result = solve(input);
    assert.equal(result, 2362);
  });
});
